Refit/RefitSettings.cs

```diff
@@ -38,32 +40,55 @@ public interface IFormUrlEncodedParameterFormatter
 
     public class DefaultUrlParameterFormatter : IUrlParameterFormatter
     {
+        static readonly ConcurrentDictionary<Type, ConcurrentDictionary<string, EnumMemberAttribute>> enumMemberCache
+            = new ConcurrentDictionary<Type, ConcurrentDictionary<string, EnumMemberAttribute>>();
+
         public virtual string Format(object parameterValue, ParameterInfo parameterInfo)
         {
             // See if we have a format
             var formatString = parameterInfo.GetCustomAttribute<QueryAttribute>(true)?.Format;
 
+            EnumMemberAttribute enummember = null;
+            if (parameterValue != null && parameterInfo.ParameterType.GetTypeInfo().IsEnum)
+            {
+                var cached = enumMemberCache.GetOrAdd(parameterInfo.ParameterType, t => new ConcurrentDictionary<string, EnumMemberAttribute>());
+                enummember = cached.GetOrAdd(parameterValue.ToString(), val => parameterInfo.ParameterType.GetMember(val).First().GetCustomAttribute<EnumMemberAttribute>());
+            }
+
             return parameterValue == null
                        ? null
                        : string.Format(CultureInfo.InvariantCulture,
                                        string.IsNullOrWhiteSpace(formatString)
                                            ? "{0}"
                                            : $"{{0:{formatString}}}",
-                                       parameterValue);
+                                       enummember?.Value ?? parameterValue);
         }
     }
 
     public class DefaultFormUrlEncodedParameterFormatter : IFormUrlEncodedParameterFormatter
     {
+        static readonly ConcurrentDictionary<Type, ConcurrentDictionary<string, EnumMemberAttribute>> enumMemberCache
```

Comment:
What is the benefit of having this static? The instance is tied to the setting object, which is tied to the interface lifetime.

Reply:
static isn't bound to any instance. The static member is thus for the lifetime of the AppDomain. (https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/static)